src/DetectorConstruction.cpp

```diff
@@ -42,25 +44,38 @@ G4VPhysicalVolume* DetectorConstruction::Construct()
   G4VPhysicalVolume* world_phys_vol =
     new G4PVPlacement(0, G4ThreeVector(0.,0.,0.),
                       world_logic_vol, "world.physical", 0, false, 0, true);
+                      
+
+  std::cout << " Detector configuration is: " << ConfigManager::GetDetectorConfiguration() << std::endl;
 
-  // DETECTOR //////////////////////////////////////////////
+  if(ConfigManager::GetDetectorConfiguration())
+  {
+  // DETECTOR HD CONFIGURATION //////////////////////////////////////////////
   // resemble an APA size
-  G4double detector_width   = 2.3*m;
-  G4double detector_height  = 6.0*m;
-  G4double detector_length  = 3.6*m;
+    ConfigManager::SetDetectorWidth(2.3*m);
+    ConfigManager::SetDetectorHeight(6.0*m);
+    ConfigManager::SetDetectorLength(3.6*m);
+  }
+  else
+  {
+  // DETECTOR VD CONFIGURATION //////////////////////////////////////////////
+    ConfigManager::SetDetectorHeight(13.0*m);
+    ConfigManager::SetDetectorLength(6.5*m);
+    ConfigManager::SetDetectorWidth(20.0*m);
+  }
```

Comment:
If you move this whole block up before you create the world volume, then you can query ConfigManager to get the largest detector dimensions, and define the world volume according to that.

Reply:
fixing in bugfix/v3.0.0_revisions branch